packages/starknet-snap/src/index.ts

```diff
@@ -208,3 +211,86 @@ export const onRpcRequest: OnRpcRequestHandler = async ({ origin, request }) =>
       throw new Error('Method not found.');
   }
 };
+
+export const onInstall: OnInstallHandler = async () => {
+  const component = panel([
+    text("Your MetaMask wallet is now compatible with Starknet!"),
+    text(`To manage your Starknet account and send and receive funds, visit the [companion dapp for Starknet](${dappUrl(process.env.SNAP_ENV)}).`),
+  ]);
+
+  await snap.request({
+    method: 'snap_dialog',
+    params: {
+      type: 'alert',
+      content: component,
+    },
+  });
+};
+
+export const onUpdate: OnUpdateHandler = async () => {
+  const component = panel([
+    text("Features released with this update:"),
+    text(
+      "**Home Page**: You can now see your account and balance on the Snap homepage.",
+    ),
+    text(
+      "To get to the Starknet Snap homepage, from the MetaMask menu, click **Snaps**, then click on the Starknet Snap."
+    ),
+  ]);
+
+  await snap.request({
+    method: 'snap_dialog',
+    params: {
+      type: 'alert',
+      content: component,
+    },
+  });
+};
+
+export const onHomePage: OnHomePageHandler = async () => {
+  const panelItems = [];
+  try {
+    const state: SnapState = await snap.request({
+      method: 'snap_manageState',
+      params: {
+        operation: 'get',
+      },
+    });
+
+    if (state && state.accContracts.length > 0) {
+      const userAddress = state.accContracts[0].address;
+      const chainId = state.accContracts[0].chainId;
+      const network = state.networks.find(n => n.chainId == chainId);
+      panelItems.push(row("Address", address(`${userAddress}`)));
+      panelItems.push(row("Network", text(`${network.name}`)));
+
+      const ercToken = state.erc20Tokens.find(t => t.symbol == "ETH" && t.chainId == chainId);
+      if (ercToken) {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        const params: any = {
+          state,
+          requestParams: {
+            tokenAddress: ercToken.address,
+            userAddress: userAddress,
+          }
+        };
+        const balance = await getErc20TokenBalance(params)
+        const displayBalance = ethers.utils.formatUnits(ethers.BigNumber.from(balance), ercToken.decimals);
+        panelItems.push(row("Balance", text(`${displayBalance} ETH`)));
+      }
+
+      panelItems.push(divider());
+      panelItems.push(text(`Visit the [companion dapp for Starknet](${dappUrl(process.env.SNAP_ENV)}) to manage your account.`));
+    } else {
+      panelItems.push(text(`**Your Starknet account is not yet deployed.**`));
+      panelItems.push(text(`Initiate a transaction to create your Starknet account. Visit the [companion dapp for Starknet](${dappUrl(process.env.SNAP_ENV)}) to get started.`));
```

Comment:
message is miss leading , losting a state doesnt means it has no account deployed

Reply:
Done